src/database.ts

```diff
@@ -206,24 +212,48 @@ export async function getPNIDProfileJSONByPID(pid: number): Promise<PNIDProfile
 }
 
 export async function getServerByGameServerID(gameServerID: string, accessMode: string): Promise<HydratedServerDocument | null> {
-	return await Server.findOne({
-		game_server_id: gameServerID,
-		access_mode: accessMode
-	});
+	const searchModes = accessModeOrder[accessMode] || ['prod'];
+
+	for (const mode of searchModes) {
+		const server = await Server.findOne({
+			game_server_id: gameServerID,
+			access_mode: mode
+		});
+
+		if (server) return server;
+	}

```

Comment:
This makes potentially three database calls

What I think we should do instead is use `Server.find` and pass the `searchModes` to the `$in` operator

We can then do this loop to find the right server

Reply:
I think I might have fixed this? Not tested since I'm in a bit of a hurry right now.

Reply:
Can confirm it works.